### ISSUE TITLE
ASTF global counter lifetime policy

### ISSUE THREAD

**Issue by egwakim** [cisco-system-traffic-generator/trex-core#374]
Hi @hhaim ,
Current ASTF global counter shows profile counters of "STOPPED" and "RUNNING" profile.
But, if we restart one of profile, then, ASTF global counter can be decreased because ASTF profile counter will be cleared when restarting same ASTF profile.

We have 2 options
Option#1. Clear global statistics counter when starting first profile in all profiles are stopped status.
ASTF global counter can be ambiguous by epoch. 
In some case, it's sum of all profiles(stopped + running) and some case, it's sum of only running profiles(counters of stopped profile will be excluded)

Option#2. Keep current policy.
ATSTF global counter can be decreased. 
But, ASTF global counter always shows sum of all profiles(stopped + running)

Both option have cons & pros
Which option do you prefer?

Best Regards
Gwangmoon

**Comment by egwakim** [comment]
Related issue is https://github.com/cisco-system-traffic-generator/trex-core/pull/367

**Comment by hhaim** [comment]
@egwakim the global counters should always go up, this is the expectation. 
so option #2 is off the table because we do want to support removing a profile while running a test with multi-profile.

There is another option to *explicitly* clear the global counter as today it is implicitly cleared when the state of ASTF global is moving  to TRANSMIT.

I don't see an issue to keep the implicit clear 

Is regard to this:
> In some case, it's sum of all profiles(stopped + running) and some case, it's sum of only running profiles(counters of stopped profile will be excluded)

I suggested a way to remove this ambiguity. My Suggestion is to keep ref in each profile and send diff to the global counter (and in case of DELETE) , this way the global counter will always go up and we won't miss any counter

So in short I would go with option #1 and I don't see any issue with it. 
Please elaborate if I'm missing something

**Comment by egwakim** [comment]
Ok,
Then, If I summarize it.
Definition of "ASTF global counter" : always increasing accumulative value ( including profile delete or all restart cases) except explicit user 'clear' command case
Required changes would be
1) New command to clear global ASTF statistics counter
2) Change behavior of global statistics counter in delete case : instead of removing, keep diff to avoid decrease

Is it correct?

**Comment by hhaim** [comment]
@egwakim 
1. Yes, just to add that we could **keep** the current functionality of explicit global counters clear -- when the ASTF global state is moving to TRANSMIT
2. There is a command to clean the globes counter today
